pkg/server/plugin/zmq/rpc.go

```diff
@@ -159,15 +159,17 @@ func (p *zmqTransport) ipc(req *pluginrequest.Request) (out []byte, err error) {
 
 	reqChan := make(chan chan []byte)
 	p.broker.RPC(reqChan, in)
+	respChan := <-reqChan
 	select {
-	case msg := <-<-reqChan:
+	case msg := <- respChan:
 		// Broker will sent back a null byte if time out
 		if bytes.Equal(msg, []byte{0}) {
 			err = fmt.Errorf("Plugin time out")
 		} else {
 			out = msg
 		}
 	case <-req.Context.Done():
+		p.broker.cancelRPC(respChan)
```

Comment:
This doesn’t seem to be robust because the context can be done and the request can finish at the same time.